File: python/src/main/python/Galvantula.py
from PokemonCharacter import PokemonCharacter
from Attack import Attack
from BugBuzz import BugBuzz
from ElectroWeb import ElectroWeb

class Galvantula(PokemonCharacter):

    def __init__(self):
        super().__init__()
        super().setType("bug/electric")
        super().setName("Galvantula")
        super().setHasEvolution(False)
        super().setSecondAttack(BugBuzz())
        super().setMainAttack(ElectroWeb())
        super().setHitPoints(77)
        super().setDefenseMultiplier(0.3)

    def evolve(self) -> str:
        return None

    def defend(self, attack:int) -> str:
        damage = int(attack * super().getDefenseMultiplier())
        newHP = super().getHitPoints() - damage
        defendMessage = "Defending attack, damage caused is {} new HP is {}".format(damage, newHP) 
        super().setHitPoints(newHP)
        return defendMessage
    
    def secondAttack(self) -> str:
        attackMessage = "Attacking opponent with {} causing a damage of {}".format(super().getSecondAttack(), super().getSecondAttackDamage())
        return attackMessage

    def mainAttack(self) -> str:
        attackMessage = "Attacking opponent with {} causing a damage of {}".format(super().getMainAttack(), super().getMainAttackDamage())
        return attackMessage
    
    def setNewAttack(self, attack:int, newAttack:Attack)->None:
        if(attack == 1):
            super().setMainAttack(newAttack)
        else:
            super().setSecondAttack(newAttack)

File: java/src/main/java/com/iteso/nintendo/Galvantula.java
package com.iteso.nintendo;

public class Galvantula extends PokemonCharacter {
    /**
     * Hit Points are the maximum life of pokemon.
     */
    public static final int HIT_POINTS = 100;
    /**
     * Defense multiplier value between 0-1.
     */
    public static final double DEFENSE_MULTIPLIER = 0.4;
    /**
     * Damage from 1 - 20.
     */
    public static final int MAIN_ATTACK_DAMAGE = 5;
    /**
     * Damage from 1 -25.
     */
    public static final int SECOND_ATTACK_DAMAGE = 15;

    /**
     * Pikachu constructor.
     */
    public Galvantula() {
        setType("electric/bug");
        setName("Galvantula");
        setHasEvolution(true);
        setSecondAttack(new BugBuzz());
        setMainAttack(new ElectroWeb());
        setHitPoints(HIT_POINTS);
        setDefenseMultiplier(DEFENSE_MULTIPLIER);
    }

    @Override
    public final String evolve() {
        return null;
    }

    @Override
    public final String defend(final int attack) {
        int damage;

        damage = (int) (attack * getDefenseMultiplier());
        int newHP = getHitPoints() - damage;

        String defendMessage = new String("Defending attack, damage caused is "
                + damage + " new HP is " + newHP);

        setHitPoints(newHP);
        return defendMessage;

    }

    @Override
    public final String secondAttack() {

        String attackMessage = new String("Attacking opponent with "
                + getSecondAttack()
                + " causing a damage of " + getSecondAttackDamage());
        return attackMessage;


    }

    @Override
    public final String mainAttack() {
        String attackMessage = new String("Attacking opponent with "
                + getMainAttack()
                + " causing a damage of " + getMainAttackDamage());
        return attackMessage;

    }

    @Override
    public final void setNewAttack(final int attack, final Attack newAttack) {
        if (attack == 1) {
            setMainAttack(newAttack);
        } else {
            setSecondAttack(newAttack);
        }
    }


}
